### ISSUE TITLE
`useExhaustiveDependencies`: False positive when destructuring props in function body

### ISSUE THREAD

**Issue by maug** [biomejs/biome#8881]
## Environment

- Biome version: Latest (tested on playground)
- Playground link: https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIAB7ACAAdQBzAGUARQBmAGYAZQBjAHQAIAB9ACAAZgByAG8AbQAgACcAcgBlAGEAYwB0ACcACgAKAHQAeQBwAGUAIABFAHgAYQBtAHAAbABlAFAAcgBvAHAAcwAgAD0AIAB7AAoACQBtAHMAZwA6ACAAcwB0AHIAaQBuAGcAOwAKAH0AOwAKAAoAYwBvAG4AcwB0ACAAXwBEAGUAcwB0AHIAdQBjAHQAdQByAGUAZABJAG4AUABhAHIAYQBtAHMAOgAgAFIAZQBhAGMAdAAuAEYAQwA8AEUAeABhAG0AcABsAGUAUAByAG8AcABzAD4AIAA9ACAAKAB7ACAAbQBzAGcAIAB9ACkAIAA9AD4AIAB7AAoAIAAgAHUAcwBlAEUAZgBmAGUAYwB0ACgAKAApACAAPQA%2BACAAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAbQBzAGcAKQAsACAAWwBtAHMAZwBdACkAOwAKAH0AOwAKAAoAYwBvAG4AcwB0ACAAXwBEAGUAcwB0AHIAdQBjAHQAdQByAGUAZABJAG4AQgBvAGQAeQA6ACAAUgBlAGEAYwB0AC4ARgBDADwARQB4AGEAbQBwAGwAZQBQAHIAbwBwAHMAPgAgAD0AIABwAHIAbwBwAHMAIAA9AD4AIAB7AAoAIAAgAGMAbwBuAHMAdAAgAHsAIABtAHMAZwAgAH0AIAA9ACAAcAByAG8AcABzAAoAIAAgAHUAcwBlAEUAZgBmAGUAYwB0ACgAKAApACAAPQA%2BACAAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAbQBzAGcAKQAsACAAWwBtAHMAZwBdACkAOwAKAH0AOwAKAA%3D%3D&ruleDomains.react=all

## Description

The `useExhaustiveDependencies` rule incorrectly reports `msg` as an unnecessary dependency when props are destructured in the function body instead of in the parameters.

## Code example

```tsx
import { useEffect } from 'react'

type ExampleProps = {
  msg: string;
};

// ✅ No warning - destructuring in parameters
const _DestructuredInParams: React.FC<ExampleProps> = ({ msg }) => {
  useEffect(() => console.log(msg), [msg]);
};

// ❌ False positive - destructuring in body
const _DestructuredInBody: React.FC<ExampleProps> = props => {
  const { msg } = props
  useEffect(() => console.log(msg), [msg]);
};
```

## Expected behavior

Both components are functionally identical. Neither should produce a warning since `msg` is:
1. Used inside the `useEffect` callback
2. Correctly listed in the dependency array

## Actual behavior

The second component (`_DestructuredInBody`) triggers:
> This hook specifies more dependencies than necessary: msg

This is incorrect - `msg` is clearly used in the effect and should be a valid dependency regardless of where the destructuring occurs.
